src/loki/reach.cc

```diff
@@ -122,5 +124,74 @@ directed_reach SimpleReach(const DirectedEdge* edge,
   return reach;
 }
 
+directed_reach Reach::operator()(const valhalla::baldr::DirectedEdge* edge,
+                                 const GraphId edge_id,
+                                 uint32_t max_reach,
+                                 valhalla::baldr::GraphReader& reader,
+                                 const std::shared_ptr<sif::DynamicCost>& costing,
+                                 uint8_t direction) {
+  // no reach is needed
+  directed_reach reach{};
+  if (max_reach == 0)
+    return reach;
+
+  max_reach_ = max_reach;
+  size_t max_labels = std::numeric_limits<decltype(reach.outbound)>::max();
+
+  // TODO: will this even work with the PBF api?
+  // fake up a location array
+  const baldr::GraphTile* tile = nullptr;
+  const auto* node = reader.GetEndNode(edge, tile);
+  auto ll = node->latlng(tile->header()->base_ll());
+
+  google::protobuf::RepeatedPtrField<Location> locations;
+  locations.Add()->mutable_ll()->set_lng(ll.first);
+  locations.Add()->mutable_ll()->set_lat(ll.second);
+  auto* path_edge = locations.Add()->add_path_edges();
+  path_edge->set_graph_id(edge_id);
```

Comment:
First segfault was due to missing edge_id (used in `SetOrigin` in `Dijkstra`. I had to propagate this in from outside. Still not sure if I did it correctly since I haven't worked on this code before.

Reply:
great catch!